pydal/objects.py

```diff
@@ -570,11 +576,18 @@ def __setitem__(self, key, value):
                 raise SyntaxError(
                     'value must be a dictionary: %s' % value)
             self.__dict__[str(key)] = value
+            if isinstance(value, (FieldVirtual, FieldMethod)):
```

Comment:
This should be moved elsewhere.
VirtualFields are defined only during table.define_table as for normal Fields.
After table.define_table neither Fields, nor VirtualFields can be defined. 

If we want so, we should introduce a new method in the table Api, something like:
table.update_fields. Which will check for all new added fields.